funcx_endpoint/tests/unit/test_endpointinterchange.py

```diff
@@ -1,29 +1,68 @@
+import random
 from unittest.mock import MagicMock
 
+import pytest
+
 from funcx_endpoint.endpoint.interchange import EndpointInterchange
+from funcx_endpoint.endpoint.utils.config import Config
+
+_mock_base = "funcx_endpoint.endpoint.interchange."
+
+
+def test_main_exception_always_quiesces(mocker, fs):
+    num_iterations = random.randint(1, 10)
```

Comment:
random iterations, but always less than the num_iterations + 10 below, right?  What's the value in sometimes failing (if the code is bad), instead of consistently failing or passing?  Why not set it just below the max allowed?

Reply:
Shoot, changed thought-process, is the reason.  That's ... I need to make this test slightly more intelligent, actually.  Good catch.

Reply:
Oh, no, I did catch it.  The reason is two-fold:

1. _This_ test is not intended to verify the `max_fail_count` behavior.  It's a previously written test to verify that we attempt to gracefully restart in the usual (error) case, so as to be able to ignore what may be spurious events.
2. I choose random as go-to testing tool in general so as to show good faith that the number chosen doesn't work by "happy coincidence."  If the code is solid, it shouldn't matter the number.  Meanwhile, I suppose we could turn this into a flat testing of all such values, but then where is a good cutoff?  So I choose random to generally cover the space.  If a test fails -- and especially in context of the maturity and skill of this group -- we'll look at that test and see what was special about the chosen number.